ui/src/components/accountSettings.js

```diff
@@ -39,18 +53,53 @@ function LogoutButton(){
 function DeactivateButton(){
     const context=useContext(AuthContext);
 
-    const deactivate=()=>{
-        
+    const [modalActive,setModalActive]=useState(false);
+    const [errorModalActive,setErrorModalActive]=useState(false);
+
+    const deactivate=(permanent)=>{
+        deactivateAccount(context.jwt,context.user.jti,permanent,(success,data)=>{
+            setModalActive(false);
+            if(success){
+                context.logout();
+                localStorage.removeItem('token');
+            }else{
+                console.log('error');
+                setErrorModalActive(true);
+            }
+        });
     }
 
     return(
         <div className="formContainer" >
+            <ReactModal isOpen={modalActive} style={{overlay:MODAL_OVERLAY,content: MODAL_CONTENT}}>
+                <div className="formContainerTitle">Deactivate account</div>
+                <div className="notSellerLabel">
+                    <p>Deactivate will deactivate account temporarly</p>
+                    <p>Delete will deactivate account permanently</p>
+                    <p>In any case all active bids will be discarded</p>
+                </div>
+                <div className="modalButtons">
+                    <span className="bidButton" onClick={()=>deactivate(false)}>Deactivate</span>
+                    <span className="bidButton" onClick={()=>deactivate(true)}>Delete</span>
```

Comment:
when passing true and false to a method like this, it is unclear what exactly does it do

it is better to have constants

deactivate(PERMANENT) \ SOFT

that way, you do not need to look at implementation of method to know what it does